pkg/archive/deb.go

```diff
@@ -74,21 +74,21 @@ func ExtractDeb(ctx context.Context, d, f string) error {
 				return fmt.Errorf("failed to close file: %w", err)
 			}
 		case tar.TypeSymlink:
-			// Skip symlinks for targets that do not exist
-			_, err = os.Readlink(target)
+			// Ensure that symlinks are not relative path traversals
+			// #nosec G305 // L88 handles the check
+			fullLink := filepath.Join(d, header.Linkname)
+			_, err = os.Lstat(fullLink)
 			if os.IsNotExist(err) {
 				continue
 			}
-			// Ensure that symlinks are not relative path traversals
-			// #nosec G305 // L208 handles the check
-			linkReal, err := filepath.EvalSymlinks(filepath.Join(d, header.Linkname))
+			linkReal, err := filepath.EvalSymlinks(fullLink)
 			if err != nil {
 				return fmt.Errorf("failed to evaluate symlink: %w", err)
 			}
-			if !IsValidPath(linkReal, d) {
+			if !IsValidPath(target, d) {
```

Comment:
I don't think this change is correct; I believe the intent here is to check that the destination of the symlink (what it points to) is within the temporary extraction directory.

However, we should also probably verifying that `isValid` is applied on the files, it seems that that's missing for the deb case:
https://github.com/chainguard-dev/malcontent/blob/3b499254a5714e4cd1cb95f05f110909cb70b849/pkg/archive/deb.go#L48-L50

unlike the tar case:

https://github.com/chainguard-dev/malcontent/blob/3b499254a5714e4cd1cb95f05f110909cb70b849/pkg/archive/tar.go#L103-L108

Reply:
Good catch -- this was definitely applied inconsistently. `zip.go` used it but I'll add it to the other methods as well.